lib/blocs/authentication/viewmodels/authentication_bloc.dart

```diff
@@ -27,14 +26,10 @@ class AuthenticationBloc extends Bloc<AuthenticationEvent, AuthenticationState>
       settingsStorage.privateKeyBackedUp = true;
       // New account --> re-start auth status
       add(const InitAuthStatus());
-      // Set fcm token must be last instruction to allow login, even if there is an error here.
-      await FirebaseMessageTokenRepository().setFirebaseMessageToken(event.account);
     }
     if (event is OnCreateAccount) {
       // New account --> re-start auth status
       add(const InitAuthStatus());
-      // Set fcm token must be last instruction to allow login, even if there is an error here.
-      await FirebaseMessageTokenRepository().setFirebaseMessageToken(settingsStorage.accountName);
     }
     if (event is EnablePasscode) {
       settingsStorage.savePasscode(event.newPasscode);
```

Comment:
the correct way to implement this is to have a use case that handles these things. 
The settingsStorage should not be doing side effect things like `_updateFirebaseToken` 

The settings storage should know only about settings.

Reply:
IDK what to do about this

Use case? And how can we make sure the use case always responds to the account name?

Then we're back in the situation someone might use that use case at the wrong time etc. 

We want to set the firebase token whenever the account name is changed. 

In reality, what we could do, is fire an event when the account name changes. That way settings doesn't need to know about firebase (which makes sense, I agree on this point)

Reply:
Ok did the use case. It's good.